nuklear.h

```diff
@@ -9358,8 +9370,10 @@ nk_draw_list_alloc_vertices(struct nk_draw_list *list, nk_size count)
      * backend (OpenGL, DirectX, ...). For example in OpenGL for `glDrawElements`
      * instead of specifing `GL_UNSIGNED_SHORT` you have to define `GL_UNSIGNED_INT`.
      * Sorry for the inconvenience. */
+#ifndef NK_UINT_DRAW_INDEX
     NK_ASSERT((sizeof(nk_draw_index) == 2 && list->vertex_count < NK_USHORT_MAX &&
```

Comment:
not really sure what you mean, the assert is only used for when NK_USHORT_MAX  is relevant.
if NK_UINT_DRAW_INDEX is set the vertex_count variable and the nk_draw_index should be the same size so it can't actually be larger.

Reply:
That's correct. There is though no connection between the declaration of `unsigned int vertex_count` in `struct nk_draw_list` and `nk_uint nk_draw_index` as `nk_uint` (defined as 32bit in case fixed types are chosen or as `unsigned int` otherwise) can differ from `unsigned int` on some platforms (can be smaller).

Reply:
there fixed the original logic instead.
if the size was not 2 it would always assert before and now it only checks if the count is over NK_USHORT_MAX  if the size == 2